### DISCUSSION TITLE
How to use audio Url in wavesurfer

### DISCUSSION THREAD

**Discussion by peterjones2023** [katspaugh/wavesurfer.js#3152] [Q&A] [answered]
When I use the audio URL(https://samplelib.com/lib/preview/mp3/sample-3s.mp3), I will get a CORS error. So I converted url audio to buffer and used this code to use and it works but in V7 I don't know how to do that because some methods are removed.
code:
var audioContext = new AudioContext();
    audioContext.decodeAudioData(buffer.buffer, function (audioBuffer) {
        wavesurfer.loadDecodedBuffer(audioBuffer);
    });

Thank you.

**Comment by katspaugh** [comment]
You can load a blob like this: https://wavesurfer-js.org/docs/classes/wavesurfer.default#loadBlob

**Reply by peterjones2023** [reply]
do we have any example of this?

**Reply by katspaugh** [reply, marked as answer]
`wavesurfer.loadBlob(new Blob([buffer]))`

**Reply by peterjones2023** [reply]
thanks if works :)

**Reply by katspaugh** [reply]
Just curious, how did you manage to fetch a buffer without CORS errors?

**Reply by peterjones2023** [reply]
when I use the audio URL in Wavesurfer in JS it will throw a CORS error. So I used C# for converting audio into buffer.
![image](https://github.com/katspaugh/wavesurfer.js/assets/135982085/d13ea31e-ade8-4684-84ba-cd552dd6cb7b)

**Reply by ManimMadhav** [reply]
hey @peterjones2023. i'm facing an issue in loading files from my S3 bucket because I keep running into CORS and ```Unable to decode audio data``` errors. could you pls guide me on how you managed to resolve these errors?
